include/swift/Threading/ThreadLocalStorage.h

```diff
@@ -115,6 +115,15 @@ class ThreadLocal {
   T get() { return value; }
 
   void set(T newValue) { value = newValue; }
+
+  T replace(T newValue) {
```

Comment:
Minor: would `swap` be a better name for this, given that we return the old value? (I'm not sure, just pondering whether there's a way to somehow make it obvious that it returns the original value.)